applications/DEMApplication/custom_conditions/RigidFace.cpp

```diff
@@ -64,59 +64,71 @@ void RigidFace3D::Initialize() {
 //***********************************************************************************
 //***********************************************************************************
 
-void RigidFace3D::CalculateRightHandSide(VectorType& rRightHandSideVector,
-                                         ProcessInfo& r_process_info)
-{
+void RigidFace3D::CalculateRightHandSide(VectorType& rRightHandSideVector, ProcessInfo& r_process_info) {
     const unsigned int number_of_nodes = GetGeometry().size();
     unsigned int MatSize = number_of_nodes * 3;
 
-    if (rRightHandSideVector.size() != MatSize)
-    {
+    if (rRightHandSideVector.size() != MatSize) {
         rRightHandSideVector.resize(MatSize, false);
     }
     rRightHandSideVector = ZeroVector(MatSize);
 
-    std::vector<SphericParticle*>& rNeighbours = this->mNeighbourSphericParticles;
+    std::vector<SphericParticle*>& vector_of_glued_particles = GetVectorOfGluedParticles();
+    for (unsigned int i=0; i<vector_of_glued_particles.size(); i++) {
+        SphericParticle* p_particle = vector_of_glued_particles[i];
+        DEMIntegrationScheme& dem_scheme = p_particle->GetTranslationalIntegrationScheme();
+        GluedToWallScheme* p_glued_scheme = dynamic_cast<GluedToWallScheme*>(&dem_scheme);
+        #ifdef KRATOS_DEBUG
+        Condition* p_condition = p_glued_scheme->pGetCondition();
+        if(p_condition != this) {
+            KRATOS_ERROR << "Inconsistency in the pointers to faces of the glued spheres!! Condition with id: " <<this->Id()<<" used a sphere with Id: "<<p_particle->Id()<<" glued to it, but the sphere was actually glued to a Condition with Id: "<<p_condition->Id()<<std::endl;
+        }
+        #endif
+        array_1d<double, 3> force = ZeroVector(3);
+        array_1d<double, 4> weights_vector= ZeroVector(4);
+        noalias(force) = p_particle->GetGeometry()[0].FastGetSolutionStepValue(TOTAL_FORCES);
+        Vector& r_shape_functions_values = p_glued_scheme->GetShapeFunctionsValues();
+        for(size_t j=0; j<r_shape_functions_values.size(); j++) {
+            weights_vector[j] = r_shape_functions_values[j];
+        }
+        for (unsigned int k=0; k< number_of_nodes; k++) {
+            rRightHandSideVector[k * 3 + 0] += force[0] * weights_vector[k];
+            rRightHandSideVector[k * 3 + 1] += force[1] * weights_vector[k];
+            rRightHandSideVector[k * 3 + 2] += force[2] * weights_vector[k];
+        }
+    }
 
-    for (unsigned int i=0; i<rNeighbours.size(); i++)
-    {
+    std::vector<SphericParticle*>& rNeighbours = this->mNeighbourSphericParticles;
+    for (unsigned int i=0; i<rNeighbours.size(); i++) {
         if(rNeighbours[i]->Is(BLOCKED)) continue; //Inlet Generator Spheres are ignored when integrating forces.
+        array_1d<double, 3> force = ZeroVector(3);
+        array_1d<double, 4> weights_vector= ZeroVector(4);
```

Comment:
Do you need to hard-code the number of weights? It is already almost general...

Reply:
Sorry I don't understand what you mean. Can you elaborate?

Reply:
Ah, done. The size is automatic now. I had to switch to std::vector because array_1d is a static vector (the std::vector will probably be slower because it's dynamic. Not really shure about this).